sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCAuthenticationService.java

```diff
@@ -0,0 +1,183 @@
+        throw new SendOtpException(ErrorConstants.NOT_IMPLEMENTED);
+        }
+
+    @Override
+    public boolean isSupportedOtpChannel(String channel) {
+        return false;
+    }
+
+    @Override
+    public List<KycSigningCertificateData> getAllKycSigningCertificates() {
+        return new ArrayList<>();
+    }
+
+    private KycAuthResult validateKnowledgeBasedAuth(String individualId, AuthChallenge authChallenge) throws KycAuthException {
+
+        KycAuthResult  kycAuthResult= new KycAuthResult();
+        RegistrySearchRequestDto registrySearchRequestDto =new RegistrySearchRequestDto();
+        registrySearchRequestDto.setLimit(2);
+        registrySearchRequestDto.setOffset(0);
+        Map<String,Map<String,String>> filter=new HashMap<>();
+        String encodedChallenge=authChallenge.getChallenge();
+
+        byte[] decodedBytes = Base64.getUrlDecoder().decode(encodedChallenge);
+        String challenge = new String(decodedBytes, StandardCharsets.UTF_8);
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(challenge);
+
+            String fieldDetails =env.getProperty("mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-detailss");
+            List<FieldDetail> fieldDetailList = objectMapper.readValue(fieldDetails, new TypeReference<List<FieldDetail>>(){});
+
+            for(FieldDetail fieldDetail:fieldDetailList){
+                Map<String,String> hashMap=new HashMap<>();
+
+                if(!StringUtils.isEmpty(idField) && fieldDetail.getId().equals(idField)){
+                    hashMap.put(fieldKey,individualId);
+
+                }else{
+                    hashMap.put(fieldKey,jsonNode.get(fieldDetail.getId()).asText());
+                }
+                filter.put(fieldDetail.getId(),hashMap);
+            }
+            registrySearchRequestDto.setFilters(filter);
+
+            String requestBody = objectMapper.writeValueAsString(registrySearchRequestDto);
+            RequestEntity requestEntity = RequestEntity
+                    .post(UriComponentsBuilder.fromUriString(registrySearchUrl).build().toUri())
+                    .contentType(MediaType.APPLICATION_JSON_UTF8)
+                    .body(requestBody);
+            ResponseEntity<List<Map<String,Object>>> responseEntity = restTemplate.exchange(requestEntity,
+                    new ParameterizedTypeReference<List<Map<String,Object>>>() {});
+            if (responseEntity.getStatusCode().is2xxSuccessful() && responseEntity.getBody() != null) {
+                List<Map<String,Object>> responseList = responseEntity.getBody();
+                if(responseList.size()==1){
+                    log.debug("getting response {}", responseEntity);
+                    kycAuthResult.setKycToken((String)responseList.get(0).get(osid));
+                    kycAuthResult.setPartnerSpecificUserToken((String)responseList.get(0).get(idField));
```

Comment:
this should also be osId as per the confluence page.

Reply:
done